spec.proto

```diff
@@ -384,6 +386,20 @@ message ProtoSpecResult {
     repeated Error errors = 10;
 }
 
+/// A proto object representing the result of Scenario execution.
+message ProtoScenarioResult {
+    /// Collection of scenarios in scenario execution result.
+    ProtoItem protoItem = 1;
+}
+
+/// A proto object representing the result of Step execution.
+message ProtoStepResult {
+    /// Collection of steps in step execution result.
+    ProtoItem protoItem = 1;
+    /// Flag to indicate if step is failed.
+    bool stepFailed  = 2;
```

Comment:
ProtoStep message has ProtoStepExecutionResult. This inturn has ProtoExecutionResult that has a flag to indicate failure. So this need not be sent explicitly.